### PR TITLE
updated nonces to bigInt

### PR DESCRIPTION
**Pull request by rauljareno**

### CODE REVIEW

File: lib/credential/domain/use_cases/get_claims_use_case.dart [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import 'package:polygonid_flutter_sdk/common/domain/domain_logger.dart';
 import 'package:polygonid_flutter_sdk/common/domain/entities/filter_entity.dart';
 import 'package:polygonid_flutter_sdk/common/domain/use_case.dart';
+import 'package:polygonid_flutter_sdk/constants.dart';
```
Comment (Plumillon) [inline lib/credential/domain/use_cases/get_claims_use_case.dart:4]: Domain cannot have dependency of the constants file.
`GENESIS_PROFILE_NONCE` needs to be in Domain.

File: lib/credential/domain/use_cases/get_claims_use_case.dart [hunk 2]
```diff
@@ -10,13 +11,13 @@ import 'package:polygonid_flutter_sdk/identity/domain/use_cases/identity/get_ide
...
   final List<FilterEntity>? filters;
   final String did;
-  final int profileNonce;
+  final BigInt profileNonce;
   final String privateKey;
 
   GetClaimsParam({
     this.filters,
     required this.did,
-    this.profileNonce = 0,
```
Comment (Plumillon) [inline lib/credential/domain/use_cases/get_claims_use_case.dart:19]: Should we let the default value as `GENESIS_PROFILE_NONCE`?

File: lib/identity/domain/use_cases/identity/backup_identity_use_case.dart [hunk 3]
```diff
@@ -1,46 +1,44 @@
...
   );
 
   @override
-  Future<Map<int, String>> execute({required String param}) async {
-    try {
-      Map<int, String> result = {};
-      String genesisDid = await _getCurrentEnvDidIdentifierUseCase.execute(
-          param: GetCurrentEnvDidIdentifierParam(privateKey: param));
+  Future<String> execute({required String param}) async {
+    String result = "";
```
Comment (Plumillon) [inline lib/identity/domain/use_cases/identity/backup_identity_use_case.dart:22]: Not used anymore.

File: lib/identity/domain/use_cases/profile/check_profile_validity_use_case.dart [hunk 4]
```diff
@@ -17,9 +17,11 @@ class CheckProfileValidityUseCase
     extends FutureUseCase<CheckProfileValidityParam, void> {
   @override
   Future<void> execute({required CheckProfileValidityParam param}) {
+    final maxVal = BigInt.parse('2') ^ BigInt.parse('248');
     return Future(() {
-      if (param.profileNonce < 0 ||
-          (param.profileNonce == 0 && param.excludeGenesis)) {
+      if (param.profileNonce.isNegative ||
+          (param.profileNonce == BigInt.zero && param.excludeGenesis) ||
```
Comment (Plumillon) [inline lib/identity/domain/use_cases/profile/check_profile_validity_use_case.dart:23]: Should it be `param.profileNonce == GENESIS_PROFILE_NONCE`?

**Review by Plumillon — CHANGES_REQUESTED** [review]
In general:
- Domain must not have dependency of `constants.dart`
- Could we keep the default value for `profileNonce` in param?

File: lib/identity/domain/use_cases/identity/backup_identity_use_case.dart (hunk 3 above)
Comment (rauljareno) [reply lib/identity/domain/use_cases/identity/backup_identity_use_case.dart:22]: resolved

File: lib/identity/domain/use_cases/profile/check_profile_validity_use_case.dart (hunk 4 above)
Comment (rauljareno) [reply lib/identity/domain/use_cases/profile/check_profile_validity_use_case.dart:23]: resolved

**Pull request merged**